### ISSUE TITLE
Display items in datagrid headers in order based on their appearance in the DOM

### ISSUE THREAD

**Issue by bashibozuk** [vmware/clarity#4189]
## Sort and filter indicators in the datagrid headers are displayed in order different form the order in the DOM which makes keyboard navigation confusing 

When a keyboard only user is navigating through the grid with the tab one is navigated first to the filter , than to the sort control but the are in the opposite order visually so it is confusing

## How to reproduce

https://v2.clarity.design/datagrid/custom-sorting

1.  Go to  https://v2.clarity.design/datagrid/custom-sorting
2.  Navigate to the first sortable and filterable column
3. Observe that first filter button is focused and than the title with the sort indicator
![grid-nav](https://user-images.githubusercontent.com/2686283/72154027-2f5f5f80-33b8-11ea-9c9b-7d9a9cf47769.gif)

## Expected behavior

The controls should be accessed with keyboard in order relevant to the order in which they are displayed, i.e. first sort , than filter
## Versions

**App**
* Clarity: 2.3.5

**Device:**

* Type: MacBook
* OS: OSX
* Browser: Chrome
* Version: 79

## Additional notes

N/A

**Comment by gnomeontherun** [comment]
Looks like a simple change of DOM order and shouldn't cause conflicts but need to verify.

**Comment by chlane** [comment]
Fix verified on https://clarity.design/documentation/datagrid/custom-sorting
